### PR TITLE
Use  ':root', rather than alias 'root' for default root collection

### LINKED ISSUE #191: Support dynamic alias for the root collection
Currently the default value of collection alias in  the Collection use cases is 'root'.  This can cause a problem if the alias of the root collection is edited.  To manage this the Dataverse API has a keyword, ':root', to get the root collection.  

Need to analyze how to update the default collection alias being used in the Collection Use cases.  For example, it makes sense to use ':root' as a default value for Get Collection, but what do we want to use as the default value for Create Collection?  

See SPA issue: https://github.com/IQSS/dataverse-frontend/issues/489

### PR DESCRIPTION
**Pull request by ekraffmiller**
## What this PR does / why we need it:
The root collection alias, 'root' is not a constant, it is an editable field in the Collection. So we need to use the special keyword ':root' to get the root collection.  This affects all use cases that default to the root collection if no collectionId param is provided.

## Which issue(s) this PR closes:

- Closes #191 

## Related Dataverse PRs:

- Depends on #

## Suggestions on how to test this:
Visual inspection of the code, including the updated Readme for the use cases, and review of the tests that use the default root collection.

### CODE REVIEW

File: src/collections/domain/useCases/GetCollection.ts [hunk 1]
```diff
@@ -12,11 +12,11 @@ export class GetCollection implements UseCase<Collection> {
   /**
    * Returns a Collection instance, given the search parameters to identify it.
    *
-   * @param {number | string} [collectionIdOrAlias = 'root'] - A generic collection identifier, which can be either a string (for queries by CollectionAlias), or a number (for queries by CollectionId)
+   * @param {number | string} [collectionIdOrAlias = ':root'] - A generic collection identifier, which can be either a string (for queries by CollectionAlias), or a number (for queries by CollectionId)
    * If this parameter is not set, the default value is: 'root'
```
Comment (g-saracca) [inline src/collections/domain/useCases/GetCollection.ts]:
```suggestion
   * If this parameter is not set, the default value is: ':root'
```

**Review by g-saracca — COMMENTED** [review]
This looks really good @ekraffmiller! 
I suggested a commit (maybe is not needed) and have a doubt about this part of the code in `collectionHelper.ts`, should be use ':root' instead of 'root' also there?
```
export async function createCollectionViaApi(
  collectionAlias: string,
  parentCollectionAlias: string | undefined = undefined
): Promise<CollectionPayload> {
  try {
    if (parentCollectionAlias == undefined) {
      parentCollectionAlias = 'root'
    }
```

Another thing, just out of curiosity, do all API endpoints work with this special id ':root'?

**Comment by ekraffmiller** [comment]
> Another thing, just out of curiosity, do all API endpoints work with this special id ':root'?



> This looks really good @ekraffmiller! I suggested a commit (maybe is not needed) and have a doubt about this part of the code in `collectionHelper.ts`, should be use ':root' instead of 'root' also there?
> 
> ```
> export async function createCollectionViaApi(
>   collectionAlias: string,
>   parentCollectionAlias: string | undefined = undefined
> ): Promise<CollectionPayload> {
>   try {
>     if (parentCollectionAlias == undefined) {
>       parentCollectionAlias = 'root'
>     }
> ```
> 
> Another thing, just out of curiosity, do all API endpoints work with this special id ':root'?

Yes, it should be `:root` thanks!
All of the endpoints should work with this id, as far as I know.  I left two use cases as is (not using :root).  For 'publishCollection', the root collection should always be published, so I didn't think it needed it.  For getCollectionItems, the API call doesn't require collectionId.

**Review by g-saracca — APPROVED** [review]

**Comment by ofahimIQSS** [comment]
Did visual inspection of code. Merging PR

**Pull request merged**